src/backend/distributed/executor/partitioned_intermediate_results.c

```diff
@@ -44,57 +44,48 @@ typedef struct PartitionedResultDestReceiver
 	/* public DestReceiver interface */
 	DestReceiver pub;
 
-	/* partition file $i is stored at file named $resultIdPrefix_$i. */
-	char *resultIdPrefix;
+	/* on lazy startup we only startup the DestReceiver once they receive a tuple */
+	bool lazyStartup;
 
-	/* use binary copy or just text copy format? */
-	bool binaryCopy;
-
-	/* used for deciding which partition a shard belongs to. */
-	CitusTableCacheEntry *shardSearchInfo;
-
-	MemoryContext perTupleContext;
-
-	/* how does stream tuples look like? */
-	TupleDesc tupleDescriptor;
+	/* cached startup information for use in lazy startup */
+	struct
+	{
+		int operation;
+		TupleDesc tupleDescriptor;
+	} startupCache;
```

Comment:
Not sure the startupCache struct adds much vs. just having 2 fields and a small comment, e.g. `(initialized during startup)`

When I see cache I kind of have the expectation that we are trying to speed something up, but we're just storing information that we need later.

Reply:
the `struct` here is to signify that these two fields are logically belonging together. Having two fields with a comment don't show their relation ship in code on use, nor is it easy to see they belong to each other, even with a comment.

The name in which it is stored might make sense to tweak a bit, something like `startupParameters` probably better tells what it is storing.